packages/exhibitor/extra/bootstrap_exhibitor_tls.py

```diff
@@ -0,0 +1,213 @@
+#!/opt/mesosphere/bin/python
+import os
+import socket
+import subprocess
+import sys
+
+from pathlib import Path
+from urllib.parse import urlparse
+
+
+TLS_ARTIFACT_LOCATION = '/var/lib/dcos/exhibitor-tls-artifacts'
+CSR_SERVICE_CERT_PATH = '/tmp/root-cert.pem'
+PRESHAREDKEY_LOCATION = '/root/.dcos-bootstrap-ca-psk'
+EXHIBITOR_TLS_TMP_DIR = '/var/lib/dcos/exhibitor/.pki'
+BOOTSTRAP_CA_BINARY = '/opt/mesosphere/bin/dcos-bootstrap-ca'
+
+
+def invoke_detect_ip():
+    try:
+        ip = subprocess.check_output(
+            ['/opt/mesosphere/bin/detect_ip']).strip().decode('utf-8')
+    except subprocess.CalledProcessError as e:
+        print("check_output exited with {}".format(e))
```

Comment:
Nit: will this really be a useful error message? Will it say something about the program that was executed? Can be tuned after merge.

Reply:
I shamelessly copied this from somewhere else in the code base.  I meant to reflector it :)